mmv1/products/documentaiwarehouse/DocumentSchema.yaml

```diff
@@ -0,0 +1,362 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'DocumentSchema'
+base_url: 'projects/{{project_number}}/locations/{{location}}/documentSchemas'
+self_link: '{{name}}'
+delete_url: '{{name}}'
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/document-warehouse/docs/manage-document-schemas'
+  api: 'https://cloud.google.com/document-warehouse/docs/reference/rest/v1/projects.locations.documentSchemas'
+description: |
+  A document schema is used to define document structure.
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  custom_import: templates/terraform/custom_import/document_ai_warehouse_document_schema.go.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "document_ai_warehouse_document_schema_text"
+    primary_resource_id: "example_text"
+    vars:
+      document_schema_name: "schema-name-text"
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'project_number'
```

Comment:
It looks like "immutable: true" is needed for the field `project_number`. 

"immutable: true" means that a new resource is created if the project_number changes.

Reply:
It is supposed to behave that way, new "Location" is needed to be provisioned if new project number is supplied